job_runner.py

```diff
@@ -0,0 +1,97 @@
+import os
+from jsonschema import validate, ValidationError
+import yaml
+from test_runner import TestRunner
+from compliance_suite.exceptions.ComplianceException import ComplianceException
+from compliance_suite.functions.colored_console import print_blue, print_red, print_green, print_yellow
+
+
+class JobRunner():
+
+    def __init__(self, tags):
+        self.path = os.getcwd()
+        self.tags = tags
+        self.test_count = 0
+        self.result = {             # To store the count of passed/failed/skipped tests
+            "passed": 0,
+            "failed": 0,
+            "skipped": 0
+        }
+        self.test_status = {        # To store the status of each test
+            "passed": [],
+            "failed": [],
+            "skipped": []
+        }
+
+    def generate_summary(self):
+
+        passed_tests = ", ".join(self.test_status["passed"])
+        failed_tests = ", ".join(self.test_status["failed"])
+        skipped_tests = ", ".join(self.test_status["skipped"])
+
+        print("\n\n\n")
+        print_yellow("{:#^90}".format("   Compliance Testing Summary   "))
+        print_yellow("#{:^88}#".format(""))
+        print_yellow("#{:^88}#".format(f"Total Tests - {self.test_count}"))
+        print_yellow("#{:^88}#".format(f'Passed - {self.result["passed"]} ({passed_tests})'))
+        print_yellow("#{:^88}#".format(f'Failed - {self.result["failed"]} ({failed_tests})'))
+        print_yellow("#{:^88}#".format(f'Skipped - {self.result["skipped"]} ({skipped_tests})'))
+        print_yellow("#{:^88}#".format(""))
+        print_yellow("{:#^90}".format(""))
+
+    def validate_job(self, yaml_data, yaml_file):
+        schema_path = os.path.join(self.path, "tests", "template", "test_template_schema.json")
+        with open(schema_path, "r") as f:
+            json_schema = yaml.safe_load(f)
+
+        try:
+            validate(yaml_data, json_schema)
+            print(f'Test YAML file valid for {yaml_file}')
+        except ValidationError as err:
+            raise ComplianceException(f"YAML schema validation error - {yaml_file}. Error details - {err.message}")
+
+    def validate_yaml(self, file, yaml_file):
+        try:
+            yaml_data = yaml.safe_load(file)
+            return yaml_data
+        except yaml.YAMLError as err:
+            raise ComplianceException(f"Invalid YAML File - {yaml_file}. Error details - {err}")
+
+    def run_jobs(self):
+
+        yaml_path = os.path.join(self.path, "tests")
+        for yaml_file in os.listdir(yaml_path):
+            if yaml_file.endswith(".yml"):
+                self.test_count += 1
+                with open(os.path.join(yaml_path, yaml_file), "r") as f:
+                    print("\n{:#^100}".format(f"     Initiating Test-{self.test_count} for {yaml_file}     "))
+                    try:
+                        yaml_data = self.validate_yaml(f, yaml_file)
+                        self.validate_job(yaml_data, yaml_file)
```

Comment:
No need to do all that with the file open. You can put the `catch` right after the YAML load call, then go back two levels of indentation once you have `yaml_data`.